src/core/pageContextProvider/__tests__/PageContextProvider.test.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import { PageContextProvider } from '../PageContextProvider';
+import { usePageContext } from '../usePageContext';
+import type { PageType } from '../../../common/headlessService/types';
+
+// A test component that consumes the context
+function TestConsumer() {
+  const context = usePageContext();
+  return <div data-testid="context-value">{JSON.stringify(context)}</div>;
+}
+
+describe('PageContextProvider', () => {
+  it('should render children components', () => {
+    render(
+      <PageContextProvider page={null}>
+        <div data-testid="child">Hello World</div>
+      </PageContextProvider>,
+    );
+
+    expect(screen.getByTestId('child')).toBeInTheDocument();
+    expect(screen.getByText('Hello World')).toBeInTheDocument();
+  });
+
+  it('should provide the context value to its children', () => {
+    const testContextValue = { uri: '/test-page' };
+    render(
+      <PageContextProvider page={testContextValue as PageType}>
+        <TestConsumer />
+      </PageContextProvider>,
+    );
+
+    const consumerElement = screen.getByTestId('context-value');
+    expect(consumerElement).toHaveTextContent(JSON.stringify(testContextValue));
+  });
+
+  it('should handle a null context value', () => {
+    render(
+      <PageContextProvider page={null}>
+        <TestConsumer />
+      </PageContextProvider>,
+    );
+
+    const consumerElement = screen.getByTestId('context-value');
+    expect(consumerElement).toHaveTextContent(JSON.stringify({ page: null }));
+  });
+
+  it('should return the initial context value when used outside a provider', () => {
+    render(<TestConsumer />);
+
+    const consumerElement = screen.getByTestId('context-value');
+    // Assuming the initial context value is { page: null }
+    expect(consumerElement).toHaveTextContent(JSON.stringify({ page: null }));
+  });
+
+  it('should update the context value when the page prop changes', () => {
+    const initialPage = { uri: '/initial-page' } as PageType;
```

Comment:
No need for " as " typecasts:
```suggestion
    const initialPage: PageType = {
      id: 'test-page-id',
      uri: '/initial-page',
    };
```